src/source/avro.rs

```diff
@@ -4,13 +4,36 @@ use metric;
 use mio;
 use serde_avro;
 use source::{TCPStreamHandler, TCP};
-use std::{io, sync};
-use std::io::Read;
+use std::{io, mem, sync};
+use std::io::{Cursor, Read};
 use util;
 
 #[derive(Default, Debug, Clone, Deserialize)]
 pub struct AvroStreamHandler;
 
+#[repr(C, packed)]
```

Comment:
Uh, hmm, packed is going to be pretty crummy on chips that demand aligned memory. I also don't see where repr C is needed? 

Seems like – and maybe I'm missing something here – we don't need either of these reprs.

Reply:
It is needed as we use `size_of` to know the size of the expected header off the wire.

Reply:
Alternatively, I can just ensure that `Header` stays aligned to u64, which would require me to either add an `unused` entry or grow `version` to a u64.

Reply:
Oh! So Avro headers are defined as being the size of whatever C's packed representation is? Got it.

Reply:
Not Avro headers, but the headers we are wrapping around it, yeah.

Reply:
Got it.